censustree/censustree.go

```diff
@@ -51,6 +51,11 @@ func New(wTx db.WriteTx, opts Options) (
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if err != nil {
```

Comment:
what err are you reading here? supposedly the one from tree.New, but that's a `:=`, so you're actually shadowing.

IMO it's much better if you avoid a named return parameter and defer func, and just call `database.Close` in the only case that matters: when `tree.New` fails.